spec/support/view_component/matchers.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Support
+  module ViewComponent
+    module Matchers
+      extend RSpec::Matchers::DSL
+
+      define :have_rendered_component do |component_class|
+        match do |rendered|
+          @stubs = _test_context.components.select { |stubbed| stubbed.instance.is_a? component_class }
+          @matching = @stubs.select { |stubbed| arguments.args_match?(*stubbed.args) }
+          @matching.any? do |stubbed|
+            rendered.content.include?(stubbed.output)
+          end
+        end
+
+        chain :with do |*args|
+          @arguments = RSpec::Mocks::ArgumentListMatcher.new(*args)
+        end
+
+      private
+
+        def arguments
+          @arguments ||= RSpec::Mocks::ArgumentListMatcher.new(any_args)
+        end
+      end
+
+      def have_rendered(object, *args)
+        return super unless object.is_a?(Class) && object < ::ViewComponent::Base
```

Comment:
What does `super` do here exactly?

Reply:
There's already a matcher `have_rendered` in some contexts (views/requests) to which I'd like to extend those matchers as well. We only want to use our matcher when component is given.